brightener.h

```diff
@@ -1,11 +1,44 @@
 #include <cstdint>
 #include "image.h"
 
-class ImageBrightener {
-    private:
-        std::shared_ptr<Image> m_inputImage;
-    public:
-        ImageBrightener(std::shared_ptr<Image> inputImage);
-        int BrightenWholeImage();
-        bool AddBrighteningImage(std::shared_ptr<Image> imageToAdd, int& attenuatedCount);
-};
+namespace ImageProcessor
+{
+
+    void ExecutePixelRunner(std::shared_ptr<Image> inputImage, const std::shared_ptr<Image>& imageToAdd, int& attenuatedCount)
+    {
+        auto BrightnerFun = [&attenuatedCount](uint8_t pixelGrayscale, uint8_t pixelGrayscaleToAdd) {
+            uint8_t outputGrayscale = pixelGrayscale;
+            if (outputGrayscale + pixelGrayscaleToAdd > 255) {
+                ++attenuatedCount;
+                outputGrayscale = 255;
+            }
+            else {
+                outputGrayscale += pixelGrayscaleToAdd;
+            }
+
+            return outputGrayscale;
+        };
+
+        inputImage->pixelRunner(BrightnerFun, imageToAdd);
+    }
+
+    bool BrightenWholeImage(std::shared_ptr<Image> inputImage, int& attenuatedCount, std::shared_ptr<Image> imageToAdd = nullptr) {
+        if (!inputImage->isSizeValid()) {
+            return false;
+        }
+
+        ExecutePixelRunner(inputImage, imageToAdd, attenuatedCount);
+
+        return true;
+    }
+
+    bool AddBrighteningImage(std::shared_ptr<Image> inputImage, const std::shared_ptr<Image>& imageToAdd, int& attenuatedCount) {
```

Comment:
Convention is clear. Both functions now return `bool` to indicate success / failure. And `attenuatedCount` by reference.
Any reason to use a `bool` return, rather than use exceptions?

Reply:
Introduced exception